docs/book/src/appendix_rust/rust-resources.md

```diff
@@ -1,17 +1,137 @@
+help you write modular ixa code.
+
+4. [Chapter 13, Section 1: Closures](https://rust-book.cs.brown.edu/ch13-01-closures.html): Ixa often
+requires the user to specify a function that gets executed in the future. This chapter goes over the
+mechanics for how anonymous functions are specified in Rust. Although the syntax is not challenging,
+this chapter shows how closures interact with other aspects of Rust, such as the ownership model,
+and therefore how you can write the most useful code that depends on closures (which ixa calls a
+callback in the documentation).
+
+In addition, [Chapter 19: Patterns](https://rust-book.cs.brown.edu/ch19-00-patterns.html) reiterates
+the power of enum variants and the `match` statement in Rust. If you find yourself writing more
+advanced Rust code, Chapters [9: Error Handling](https://rust-book.cs.brown.edu/ch09-00-error-handling.html),
+[18: Object-oriented programming](https://rust-book.cs.brown.edu/ch18-00-oop.html), and
+[20: Advanced Features](https://rust-book.cs.brown.edu/ch20-00-advanced-features.html) include
+helpful tips as you dive into more complicated ixa development.
+
+If you find yourself writing more analysis-focused code in Rust, Chapters [9: Error Handling](https://rust-book.cs.brown.edu/ch09-00-error-handling.html) and [13.2: Iterators](https://rust-book.cs.brown.edu/ch13-02-iterators.html)
+include helpful tools for writing code reminiscent of Python.
+
+### On Ownership
+
+Rust's signature feature is its ownership rules. Many of the details of ownership are taken care of
+by the ixa framework, meaning ixa users are often abstracted from Rust ownership challenges.
+Nevertheless, understanding ownership rules is valuable for debugging more complicated ixa code and
+understanding what you can and cannot do in Rust. There are excellent resources for understanding
+Rust's ownership rules available [online](https://educatedguesswork.org/posts/memory-management-4/),
+but at a high-level, here are the key ideas to understand:
+
+1. When a function takes an object as input, it is taking ownership of that object -- meaning that
+the object is "consumed" by the function and does not exist after the calling of that function.
+    - This is true for all types except those that implement the `Copy` trait. All primitive types --
+    such as floats, integers, booleans, etc. -- implement the `Copy` trait, meaning that this
+    rule is really only felt with vectors and hashmaps.
+2. References (`&Object`) allow functions to have access to the object without taking ownership of it.
+Therefore, it's often useful to return a reference to an object without giving up ownership of the
+object, such as when there's data that's owned by `Context`.
+3. There are two kinds of references -- mutable references `&mut Object` and immutable references
+`&Object`. There is no limit to the number of immutable references to an object that can be active
```

Comment:
This is a little unclear here – I think you should also emphasize you can either have multiple immutable references OR a single mutable reference, but not both